src/regdesk/communication_manager.js

```diff
@@ -73,6 +84,48 @@ export class CommunicationManager {
     return results;
   }
 
+  /**
+   * Handle a request from the extension to print a label.
+   * @param {*} labelDetails - The label details to pass to a BadgeLabelBuilder.
+   * @return {*} Result object of the print request.
+   */
+  async #handleExtensionPrintRequest(labelDetails) {
+    if (!labelDetails) {
+      return { success: false };
+    }
+
+    const label = new BadgeLabelBuilder(labelDetails);
+
+    try {
+      await this.#printerMgr.printLabelBuilder(label);
+    } catch {
+      return { success: false };
+    }
+
+    return { success: true };
+  }
+
+  /**
+   * Handle messages from other extension components.
+   * @param {*} o - The request object.
+   * @param {string} o.type - The request type.
+   * @param {*} o.payload - The request payload.
+   * @param {*} sender - The sender of the event.
+   * @param {function(*)} callback - Callback message to response to the event sender.
+   * @return {boolean} - True to indicate this will response async.
+   */
+  #handleExtensionMessage({ type, payload }, sender, callback) {
+    console.debug('Received extension message', type, payload, sender);
+
+    switch (type) {
+    case MESSAGE_TYPE.printLabel:
+      this.#handleExtensionPrintRequest(payload?.labelDetails).then(callback);
+      break;
+    }
+
+    return true;
```

Comment:
Do we need to trigger the callback manually when the switch/case doesn't match?

Reply:
We don't necessarily know what the callback was for or what data it wants. Safer I would think to just drop it on the floor and hope that upstream has some kind of way to tell it didn't get through.

Reply:
That's fair. Should we only return true in the switch case then?

Reply:
The 'true' here only indicates that we want to respond asynchronously to the incoming message, it doesn't get passed to the actual sender. Either way (async or not) the sender won't receive a response.

It's possible we could leak _something_ by saying we'll respond async but don't, but the docs don't seem to indicate it's a problem? I'll keep an eye out for weirdness during testing to see if anything comes of it.